src/fields/field_fe.cc

```diff
@@ -646,7 +646,7 @@ void FieldFE<spacedim, Value>::calculate_native_values(ElementDataCache<double>:
 	else mesh = dh_->mesh();
 	for (auto ele : mesh->elements_range()) { // remove special case for rank == 0 - necessary for correct output
 		if (this->boundary_domain_) dof_size = value_handler1_.get_dof_indices( ele, dof_indices_ );
-		else dof_size = dh_->get_loc_dof_indices( ele, dof_indices_ );
+		else dof_size = dh_->cell_accessor_from_element(ele.idx()).get_loc_dof_indices( dof_indices_ );
```

Comment:
I guess using cell_accessor_from_element() is a temporary solution. In fact, we should iterate using dh_->local_range(), which in the case of sequential DH includes all elements. See also other places in the pull request.

Reply:
Yes this is temporary. Reason are fields with boundary domain. One possible solution is to have sort of restriction of a DH to the boundary. Then FieldFE could be agnostic about its domain.